### PR TITLE
Make messages from influx more often in `alerta`

### PR DESCRIPTION
**Pull request by lego963**
Closes: #96

### CODE REVIEW

File: delatore/sources/base.py [hunk 1]
```diff
@@ -69,7 +70,9 @@ async def start(self, stop_event: asyncio.Event):
                 new = await asyncio.wait_for(self.get_update(), self.request_timeout)
             except (asyncio.TimeoutError, NoUpdates):
                 continue
-            if _same_status(new, last) and self.ignore_duplicates:
+            if self.ignore_duplicates and \
+                    same_status(new, last) and \
+                    _heartbeat_status(new, last) < self.heartbeat_interval:
```
Comment (outcatcher) [inline delatore/sources/base.py:75]: `_heartbeat_status(new, last) < self.heartbeat_interval` can be part of `same_status`

Comment (lego963) [reply delatore/sources/base.py]: no, it needs to be different checks

File: delatore/configuration/resources/sources.yaml [hunk 2]
```diff
@@ -29,22 +29,28 @@
         metric_id: diskio
         query: "SELECT LAST(writes) FROM {entity} WHERE host=~/scn3/ LIMIT 1;"
         timeout: 300
-  publishes: 'INFLUX'
+  publishes:
+    - 'INFLUX_tg'
```
Comment (outcatcher) [inline delatore/configuration/resources/sources.yaml:33]: Topic should be less specific about it future subscribers, e.g. `INFLUX_INFO`, `INFLUX_DEBUG` or `INFLUX_ALL`, `INFLUX_CHANGES`

File: delatore/configuration/resources/sources.yaml [hunk 3]
```diff
@@ -29,22 +29,28 @@
...
+    - 'AWX_CLIENT_OUT_al'
   timings:
     polling_interval: .1  # /status request polling
     request_timeout: 5
 - name: awx_web_hook
   params:
     port: 23834
-  publishes: 'AWX_WEB_HOOK'
+  publishes:
+    - 'AWX_WEB_HOOK_tg'
```
Comment (outcatcher) [inline delatore/configuration/resources/sources.yaml:52]: Same as in previous

File: delatore/sources/base.py [hunk 4]
```diff
@@ -26,15 +26,17 @@ class SourceMeta(ABCMeta):
...
+    TOPIC_al: str
     config: SourceConfiguration
 
     def __init__(cls, name, bases=(), dct=None):  # pylint:disable=redefined-builtin
         super().__init__(name, bases, dct)
         if not isabstract(cls):  # skip configuration for abstracts
             assert hasattr(cls, 'CONFIG_ID')
             cls.config = SOURCES_CFG[cls.CONFIG_ID]
-            cls.TOPIC = cls.config.publishes
+            cls.TOPIC_tg = cls.config.publishes[0]
```
Comment (outcatcher) [inline delatore/sources/base.py:38]: If the number of topics is fixed, topics should be a structure, not list

File: delatore/sources/base.py [hunk 5]
```diff
@@ -69,16 +72,31 @@ async def start(self, stop_event: asyncio.Event):
...
             json_message = json.dumps(new)
-            await self.client.publish(self.TOPIC, json_message)
+            await self.client.publish(self.TOPIC_tg, json_message)
+            await self.client.publish(self.TOPIC_al, json_message)
             last = new
             LOGGER.debug('Wait for new data for %s', self.polling_interval)
             await asyncio.sleep(self.polling_interval)
 
 
+def _heartbeat_status(new: dict, last: dict) -> int:
```
Comment (outcatcher) [inline delatore/sources/base.py:94]: How can `status` return int?

File: delatore/sources/base.py [hunk 6]
```diff
@@ -69,16 +72,31 @@ async def start(self, stop_event: asyncio.Event):
...
+                last = new
+                continue
+            elif self.ignore_duplicates and \
+                    _same_status(new, last):
                 continue
             LOGGER.debug('New data received from source: %s\ndata:\n%s', name, new)
             json_message = json.dumps(new)
-            await self.client.publish(self.TOPIC, json_message)
+            await self.client.publish(self.TOPIC_tg, json_message)
+            await self.client.publish(self.TOPIC_al, json_message)
```
Comment (outcatcher) [inline delatore/sources/base.py:88]: Always publishes to both topics

File: delatore/sources/base.py [hunk 7]
```diff
@@ -22,19 +22,31 @@ class NoUpdates(Exception):
...
+        prefix = prefix.upper()
+        return Topics(f'{prefix}_CHANGES', f'{prefix}_INFO')
+
+
 class SourceMeta(ABCMeta):
     """Source class meta setting up data from configuration"""
 
     CONFIG_ID: str
-    TOPIC: str
+    PREFIX: str
```
Comment (outcatcher) [inline delatore/sources/base.py:39]: Is prefix needed afterward?

File: delatore/sources/base.py [hunk 8]
```diff
@@ -69,16 +82,30 @@ async def start(self, stop_event: asyncio.Event):
                 new = await asyncio.wait_for(self.get_update(), self.request_timeout)
             except (asyncio.TimeoutError, NoUpdates):
                 continue
-            if _same_status(new, last) and self.ignore_duplicates:
+            if self.ignore_duplicates and \
+                    _same_status(new, last) and \
+                    _heartbeat_timestamp(new, last) >= self.heartbeat_interval:
+                json_message = json.dumps(new)
+                await self.client.publish(self.TOPICS.info, json_message)
```
Comment (outcatcher) [inline delatore/sources/base.py:89]: Should it be here? If not `_same_status(new, last)`, no info messagge will be published

File: delatore/sources/base.py [hunk 9]
```diff
@@ -69,16 +82,30 @@ async def start(self, stop_event: asyncio.Event):
...
             LOGGER.debug('New data received from source: %s\ndata:\n%s', name, new)
             json_message = json.dumps(new)
-            await self.client.publish(self.TOPIC, json_message)
+            await self.client.publish(self.TOPICS.changes, json_message)
             last = new
             LOGGER.debug('Wait for new data for %s', self.polling_interval)
             await asyncio.sleep(self.polling_interval)
 
 
+def _heartbeat_timestamp(new: dict, last: dict) -> int:
```
Comment (outcatcher) [inline delatore/sources/base.py:103]: maybe `delta_seconds`?

File: tests/02_acceptance/21_sources_ext/test_awx_web_hook.py [hunk 10]
```diff
@@ -29,6 +29,7 @@ async def test_http_server_notification(awx_web_hook_src: AWXWebHookSource, sub:
             pass
     assert resp.status == 200
     data = json.loads(await sub.get(.5))
+    del data['message_timestamp']  # stub for test
```
Comment (outcatcher) [inline tests/02_acceptance/21_sources_ext/test_awx_web_hook.py:32]: You can use `data.pop('message_timestamp')` here

File: delatore/sources/base.py (hunk 8 above)
Comment (lego963) [reply delatore/sources/base.py:89]: If not `_same_status(new, last)` it will publish in changes

File: delatore/sources/base.py [hunk 11]
```diff
@@ -22,19 +22,29 @@ class NoUpdates(Exception):
     """Raised when source has no updates"""
 
 
+class Topics(NamedTuple):
+    changes: str
+    info: str
+
+    @classmethod
+    def get_prefix(cls, prefix: str):
```
Comment (outcatcher) [inline delatore/sources/base.py:30]: `get_prefix` is when you return prefix, in this case it is `from_prefix`, `with_prefix`, etc.

File: delatore/sources/base.py [hunk 12]
```diff
@@ -69,16 +80,27 @@ async def start(self, stop_event: asyncio.Event):
                 new = await asyncio.wait_for(self.get_update(), self.request_timeout)
             except (asyncio.TimeoutError, NoUpdates):
                 continue
-            if _same_status(new, last) and self.ignore_duplicates:
+            if self.ignore_duplicates and \
```
Comment (outcatcher) [inline delatore/sources/base.py:83]: No need to break the line into two

**Pull request merged**